fastseq_cli/transformers_generate.py

```diff
@@ -17,6 +18,37 @@ def chunks(lst, n):
     for i in range(0, len(lst), n):
         yield lst[i:i + n]
 
+def sort_sentences(input_ids: List[str], reverse: bool=False):
+    """Sort the tokenized sentences by the number of tokens.
+
+    Args:
+        input_ids (List[List[int]]): tokenized sentences.
+        reverse (bool): indicate the order is ascending(False) or descending.
+
+    Returns:
+        tuple(List[List[int]], List[int]): the sorted tokenized sentences and
+            the indices in the original input list.
+    """
+    is_ascending = -1 if reverse else 1
+    sorted_idx = sorted(
+        range(len(input_ids)), key=lambda i: len(input_ids[i])*is_ascending)
+    sorted_input_ids = [input_ids[i] for i in sorted_idx]
+    return sorted_input_ids, sorted_idx
+
+def unsort_sentences(sents: List[str], sorted_idx: List[int]):
+    """Unsort the sents to be the order specified by sorted_idx.
+
+    Args:
+        sents (List[str]): a list of input strings.
+        sorted_idx (List[int]): the order that will be restored.
+
+    Returns:
+        List[str]: the unsorted list of strings.
+    """
+    result = [''] * len(sents)
+    for cur_idx, org_idx in enumerate(sorted_idx):
+        result[org_idx] = sents[cur_idx]
+    return result
```

Comment:
it will double memory usage.
To avoid it, how about sort by original_idx, then print according to it?

Reply:
I use `ascending` at the beginning, but found `reverse` is used by python built-in [sorted(...)](https://docs.python.org/3/library/functions.html#sorted) function. Which one do you think is better?

Reply:
do you mean there is a naming conflict?

Reply:
No, there is no conflict. We can use any names here. 

I mean do we need to use the same parameter name as `sorted(...)` uses?

Reply:
> it will double memory usage.
> To avoid it, how about sort by original_idx, then print according to it?

@JiushengChen , the memory will not be doubled. Here is copied by reference. It is confirmed by here (https://github.com/microsoft/fastseq/pull/66/commits/aa174a8fb43e8d4abdb370f786e6f93303f86ea7).